src/concrete_solve.jl

```diff
@@ -61,10 +61,10 @@ function DiffEqBase._concrete_solve_adjoint(prob,alg,
 
   # Capture the callback_adj for the reverse pass and remove both callbacks
   kwargs_adj = NamedTuple{Base.diff_names(Base._nt_names(values(kwargs)), (:callback_adj,:callback))}(values(kwargs))
-
+  isq = sensealg isa QuadratureAdjoint
   if typeof(sensealg) <: BacksolveAdjoint
     sol = solve(_prob,alg,args...;save_noise=true,save_start=save_start,save_end=save_end,saveat=saveat,kwargs_fwd...)
-  elseif ischeckpointing(sensealg)
+  elseif ischeckpointing(sensealg) || (cb!==nothing && !isq)

```

Comment:
This was needed to make the first two InterpolatingAdjoint tests work. Without this, sol.ts won't be given by saveat + affect times (but by the adaptively chosen time steps).